pybossa/api/__init__.py

```diff
@@ -417,6 +420,80 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_user_preferences(user_name=None):
```

Comment:
is this (and update) api request to be for admin/subadmin, suitable access control can be applied.

Reply:
The method can currently be called by any user to update their own task preference via API, or for sub-admins/admins to update task preferences for other users via API.

Reply:
Added permission check in `get` method. https://github.com/bloomberg/pybossa/pull/675/commits/f31f16c0ce7b9e1cfdc1bab52f5197da655bc3fb

Reply:
regular users (non admin, non subadmin) cannot update `profile` and `user_type` under their pref. Are these user attributes to be dropped from response for such users?

Reply:
The method `get_user_pref_metadata()` allows workers to update their own profile (it returns True if the user is trying to update their own profile). Should they be explicitly prevented from updating and only allow admin type users?